### ISSUE TITLE
error with clojure 1.7.0

### ISSUE THREAD

**Issue by trevor** [clj-jgit/clj-jgit#34]
See discussion here: https://groups.google.com/d/msg/clojure/fb-oBpcP2yU/M0h3fmMALoMJ

```clj
#error {
 :cause Unable to resolve symbol: private-key in this context
 :via
 [{:type clojure.lang.Compiler$CompilerException
   :message java.lang.RuntimeException: Unable to resolve symbol: private-key in this context, compiling:(clj_jgit/porcelain.clj:482:40)
   :at [clojure.lang.Compiler analyze Compiler.java 6543]}
  {:type java.lang.RuntimeException
   :message Unable to resolve symbol: private-key in this context
   :at [clojure.lang.Util runtimeException Util.java 221]}]
 :trace
 [[clojure.lang.Util runtimeException Util.java 221]
  [clojure.lang.Compiler resolveIn Compiler.java 7029]
  [clojure.lang.Compiler resolve Compiler.java 6973]
  [clojure.lang.Compiler analyzeSymbol Compiler.java 6934]
  [clojure.lang.Compiler analyze Compiler.java 6506]
  [clojure.lang.Compiler analyze Compiler.java 6485]
  [clojure.lang.Compiler$HostExpr$Parser parse Compiler.java 963]
```

**Comment by magnars** [comment]
Any chance of having this resolved? I'd be happy to whip up a Pull Request if that's any help. @dsabanin

**Comment by dsabanin** [comment]
Hey! Sorry, didn't get a chance to fix this yet. If you can help with a pull request that would be greatly appreciated! Thank you.

**Comment by magnars** [comment]
Referencing destructured names from `:keys` in the `:or` clause no longer seems to work, so I had to introduce a `let` instead. Hope this looks okay, @dsabanin.

**Comment by magnars** [comment]
Seems like you already had a pull request to fix this issue in #35. I like @lorddoig's fix a little better, since it skips the `let` and just inlines the `or`. Feel free to use that one instead. :) I've run the tests with Clojure 1.7.0-beta3 along with this fix, and they all pass.

**Comment by magnars** [comment]
Any chance this could be merged and released, @dsabanin ? I'm waiting on it to upgrade. :)

**Comment by dsabanin** [comment]
Hi! Thank you for you work on this and sorry for wasting your time working on the fix that was already done. I applied #35. It's released as version 0.8.8 on Clojars. Please let me know if you have any issues running the new version. Thanks for contributing.

**Comment by magnars** [comment]
No worries. Thanks for cutting the release. I'm now on Clojure 1.7 :)
